src/components/login-button.tsx

```diff
@@ -0,0 +1,32 @@
+import { getServerAuthSession } from "~/server/auth";
+
+export default async function LoginButton() {
+  const session = await getServerAuthSession();
+
+  if (!session) {
+    return (
+      <a
```

Comment:
A neat trick to by-pass the `next-auth` page and go directly to the provider would be to do something like the following (essentially calling a function instead of redirecting to a different URL):

```javascript
"use client";

import { Button } from "~/components/ui/button";
import { signIn } from "next-auth/react";
export default function LoginButton() {
  return (
    <Button onClick={() => signIn("google")}>
    // ...
  );
}
```

Add the relevant Tailwind classes to make the button look good, of course. This _might_ break Vercel preview since we don't use Google OAuth for the previews so let me know if that happens and we can look into ways of circumventing that.